python/ray/train/tests/test_base_trainer.py

```diff
@@ -166,6 +165,29 @@ def check_override(self):
     tune.run(trainer.as_trainable(), config=new_config)
 
 
+def test_reserved_cpus(ray_start_4_cpus):
+    def train_loop(self):
+        ray.data.range(10).show()
+
+    # Will deadlock without reserved CPU fractin.
+    scale_config = {"num_workers": 1, "_max_cpu_fraction_per_node": 0.9}
+    trainer = DummyTrainer(
+        train_loop,
+        scaling_config=scale_config,
+    )
+    tune.run(trainer.as_trainable(), num_samples=4)
+
+
+# TODO(ekl/sang) this currently fails.
```

Comment:
@rkooo567 , it seems this fails since all CPUs end up excluded. Can we ensure at least 1 CPU is available on either side no matter how aggressive the fraction is?

Btw, I think we should disallow 1.0 and 0.0 as values (raise ValueError).

Reply:
Filed https://github.com/ray-project/ray/issues/26635